### PR TITLE
Allow substitution of user-defined variables in RPM preamble

### PR DESCRIPTION
**Pull request by kellyma2**
It's desirable to be able to parameterize some variables in the preamble such as architecture when RPM packages.  This change enables variable substitution in the preamble section so that the values may be injected in this fashion in lieu of only using statically defined values.

### CODE REVIEW

**Review by cgrindel — COMMENTED** [review]
Overall, the changes look reasonable to me. Could you ping me when CI is green?

**Comment by kellyma2** [comment]
Will do!

File: pkg/private/util.bzl [hunk 1]
```diff
@@ -82,5 +82,14 @@ def substitute_package_variables(ctx, attribute_value):
 
     # Map $(var) to {x} and then use format for substitution.
     # This is brittle and I hate it. We should have template substitution
-    # in the Starlark runtime.
-    return attribute_value.replace("$(", "{").replace(")", "}").format(**vars)
+    # in the Starlark runtime.  This loop compensates for mismatched counts
+    # of $(foo) so that we don't try replace things like (bar) because we
+    # have no regex matching
+    for _ in range(attribute_value.count("$(")):
+        if attribute_value.find(")") == -1:
```
Comment (aiuto) [inline pkg/private/util.bzl:89]: This looks like it forces you to use the $(foo) format rather than ${var}.

**Review by aiuto — COMMENTED** [review]
The PR description doesn't seem to match the code at all.  It looks like the code just adds some safety against someone mis-typing  a variable like  $(foo} something (bar) .

**Comment by kellyma2** [comment]
> The PR description doesn't seem to match the code at all. It looks like the code just adds some safety against someone mis-typing a variable like $(foo} something (bar) .

The initial PR + first commit match up.  When the CI job ran to test the initial changes, it hit the former case that the second commit fixes.

File: pkg/private/util.bzl (hunk 1 above)
Comment (kellyma2) [reply pkg/private/util.bzl:89]: No, it's doing something slightly different.  The original line:

```
attribute_value.replace("$(", "{").replace(")", "}").format(**vars)
```

replaces every instance of "$(" with "{" and ")" with "}" in a single shot.  The problem with this is that we'll do things like transform `foo)` into `foo}` or `(foo)` into `{foo}`.  The former is probably (mostly) harmless, but the latter results in `str.format(**vars)` thinking that there's an additional format var in there while there won't be a matching one in `vars`, generating a bit of a weird error.  Normally I'd have dealt with this using regex matching, but that isn't an option here :)

**Review by aiuto — APPROVED** [review]

**Pull request merged**